examples/cloning.rs

```diff
@@ -1,14 +1,12 @@
 //! This example demonstrates using the [ColumnBatch][hecs::ColumnBatch] API to efficiently clone
 //! the entities in a [World] along with some or all components.
 //!
-//! Note that the cloned world may have different iteration order and/or newly created entity ids
-//! may diverge between the original and newly created worlds. If that is a dealbreaker for you,
-//! see https://github.com/Ralith/hecs/issues/332 for some pointers on preserving entity allocator
-//! state; as of time of writing, you'll need to patch `hecs`.
-
-use std::any::TypeId;
+//! By using [`World::set_freelist()`], we ensure that the cloned world will allocate new entities
+//! in sync with the original world: newly allocated entities will have the same entity ids as the
+//! original world, and the iteration order will be the same.
```

Comment:
I don't think this is enough to guarantee consistent iteration order. IIRC that depends on the order in which archetypes are introduced, which might still vary.

Reply:
Actually, this probably does work because the example is saving/restoring column batches in order. Probably merits calling that out explicitly.

Reply:
Done, by updating the docs and adding a comment on the impl explaining why iteration order is incidentally preserved by this approach.